### PR TITLE
Merging features/NullableReferenceTypes to dev16.0.x

### PR DESCRIPTION
**Pull request by jcouv**
Note: this PR should not be merged as-is. 
We need PR https://github.com/dotnet/roslyn/pull/30312 to be merged first (removing last PROTOTYPE marker).

### CODE REVIEW

File: .vsts-dotnet.yml [hunk 1]
```diff
@@ -0,0 +1,129 @@
+phases:
```
Comment (jcouv) [inline .vsts-dotnet.yml]: @jaredpar @cston I'm surprised to see changes not related to nullable in this merge. Is this expected?

File: docs/compilers/CSharp/Compiler Breaking Changes - VS2019.md [hunk 2]
```diff
@@ -1,23 +1,10 @@
-**This document lists known breaking changes in Roslyn vNext (VS2019) from Roslyn 2.9 (VS2017) and native C# compiler (VS2013 and previous).**
+**This document lists known breaking changes in Roslyn 3.0 (Visual Studio 2019) from Roslyn 2.* (Visual Studio 2017)
```
Comment (jcouv) [inline docs/compilers/CSharp/Compiler Breaking Changes - VS2019.md]: This feels like we had a bad merge :-(

File: Compilers.sln [hunk 3]
```diff
@@ -22,6 +22,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.CodeAnalysis.CSha
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.CodeAnalysis.CSharp.Emit.UnitTests", "src\Compilers\CSharp\Test\Emit\Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.csproj", "{4462B57A-7245-4146-B504-D46FDE762948}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.CodeAnalysis.CSharp.IOperation.UnitTests", "src\Compilers\CSharp\Test\IOperation\Microsoft.CodeAnalysis.CSharp.IOperation.UnitTests.csproj", "{1AF3672A-C5F1-4604-B6AB-D98C4DE9C3B1}"
```
Comment (jcouv) [inline Compilers.sln]: This change also seems unexpected. Should I remove it?

Comment (cston) [reply Compilers.sln]: One unit test project was split into two in the feature branch. In short, this new project is expected.

File: .vsts-dotnet.yml (hunk 1 above)
Comment (jcouv) [reply .vsts-dotnet.yml]: Per discussion with Jard, I'll remove this file

File: Compilers.sln (hunk 3 above)
Comment (jcouv) [reply Compilers.sln]: Jared said that is expected

File: docs/compilers/CSharp/Compiler Breaking Changes - VS2019.md (hunk 2 above)
Comment (jcouv) [reply docs/compilers/CSharp/Compiler Breaking Changes - VS2019.md]: Fixed in https://github.com/dotnet/roslyn/pull/30312

**Comment by jcouv** [comment]
FYI @jasonmalinowski @jaredpar Merged

**Pull request merged**